### DISCUSSION TITLE
SOQL extension: Retrieving more than 2000 rows

### DISCUSSION THREAD

**Discussion by Andrei2233** [forcedotcom/salesforcedx-vscode#5373] [Ideas]
Awesome extension by the way!

Currently, there's a limit of retrieving max 2000 rows. It would be awesome if this could be greatly increased. 🙈

**Comment by diyer** [comment]
Thanks for the suggestion @Andrei2233 ! Converting this to a discussion for our community

**Comment by mcoTLC** [comment]
I'll add to this - the solution seems like adding an "Offset" field (similar to the "Limit" field) would solve this. It would, as with the "LIMIT" function, add a line to the end of the query to "OFFSET" instead of "LIMIT" (or along with, if the user desires).

**Comment by daphne-sfdc** [comment]
Good idea - GUS WI here: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002YillWYAR/view

**Comment by daphne-sfdc** [comment]
Hi @Andrei2233, We have some good news for you!  In [this PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/7261), we added a new `Max Query Limit` setting to allow users to change the value of the maximum number of rows they can retrieve when running SOQL queries.  The new default number of rows returned has also been increased to 50000. 🙂 

This feature was released in v66.11.0 of the SOQL extension.  Update to the latest version of the extensions to try it out! 💃
